cli/azd/pkg/tools/azcli/spring_app.go

```diff
@@ -93,7 +93,8 @@ func (ss *springService) GetSpringAppProperties(
 
 	var fqdn []string
 	if springApp.Properties != nil &&
-		springApp.Properties.Fqdn != nil {
+		springApp.Properties.Fqdn != nil &&
+		*springApp.Properties.Public {
```

Comment:
Just double checking here: is ASA guaranteed to return `true` or `false` and not a value that'd be treated as `nil`: (in JSON terms: `null`, `"null"`, `""`, etc.?

Reply:
We have been tested it in the ASA template, the `Public` property will return `true` or `false`, and no other value.